projboard/tests/test_article.py

```diff
@@ -0,0 +1,148 @@
+import pytest
+from projboard.models.article import Article, Like, View, Subject, User
+
+
+TITLE = "Article Test"
+CONTENT = "I'm not gonna write all that!"
+
+
+class TestArticle:
+
+    @pytest.fixture
+    def sup(self):
+        return Subject(name='N')
+
+    @pytest.fixture
+    def sup1(self):
+        return Subject(name='R')
+
+    @pytest.fixture
+    def user(self):
+        return User(name="Anas", nickname="A99", email="Anas@gmail.com", password="Anas123")
+
+    @pytest.fixture
+    def art(self, user, sup):
+        return Article(user_id=user, title=TITLE, content=CONTENT, subject_id=sup)
+
+    # @pytest.mark.skip
+    def test_objectCreation(self, user, art):
+        assert art.title == TITLE
+        assert art.content == CONTENT
+        # testing __str__ function
+        assert art.__str__() == art.title
+
+    # @pytest.mark.skip
+    def test_PartFunctions(self, art):
+        # testing @property functions => PartOf..()
+        # whent title & content length is less than reality
+        assert art.PartOfTitle(5) == art.title[:5] + '...'
+        assert art.PartOfContent(5) == art.content[:5] + '...'
+        # whent title & content length is greater than reality
+        assert art.PartOfTitle(100) == art.title[:100]
+        assert art.PartOfContent(100) == art.content[:100]
+
+    # @pytest.mark.skip
+    @pytest.mark.django_db
+    def test_objectSavedInDb(self, user, art, sup):
+        sup.save()
+        user.save()
+        art.save()
+        assert art in Article.objects.all()
+
+    # @pytest.mark.skip
+    @pytest.mark.django_db
+    # using Edite function
+    def test_EditeObject(self, user, art, sup, sup1):
+        sup.save()
+        sup1.save()
+        user.save()
+        art.save()
+
+        title = "New Title"
+        content = "New Content"
+        subject = sup1
+
+        assert art.subject_id == sup
+
+        art.Edite(title, content, subject)
+
+        assert art.title == title
+        assert art.content == content
+        assert art.subject_id == sup1
+        # if subject not in Article.SUBJECTS the value won't be changed
+        subject = Subject(name='k')
+        art.Edite(subject=subject)
+        assert art.subject_id == sup1
+
+    @pytest.fixture
+    def user1(self):
+        return User(name="John", nickname="J99", email="John@gmail.com", password="John123")
+
+    @pytest.fixture
+    def art1(self, user1, sup1):
```

Comment:
Please return in one fixture a list,
Could be one or more

Reply:
done